### PR TITLE
Teams support xsoar8

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9036

## Description
added support to xsoar 8.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md [hunk 1]
```diff
@@ -56,18 +56,21 @@ In order to verify that the messaging endpoint is open as expected, you can surf
 ### 1. Using Cortex XSOAR rerouting
 In this configuration, we will use Cortex XSOAR functionality, which reroutes HTTPS requests that hit the default port (443) to the web server that the integration spins up.
 
-The messaging endpoint needs to be: `<CORTEX-XSOAR-URL>/instance/execute/<INTEGRATION-INSTANCE-NAME>`, e.g., `https://my.demisto.live/instance/execute/teams`
+The messaging endpoint needs to be:
+
+For XSOAR version less than 8.0.0: `<CORTEX-XSOAR-URL>/instance/execute/<INTEGRATION-INSTANCE-NAME>`, e.g., `https://my.demisto.live/instance/execute/teams`.
+
+For XSOAR version 8.0.0. and greater: `https://ext-<CORTEX-XSOAR-Domain>/xsoar/instance/execute/<INTEGRATION-INSTANCE-NAME>`, e.g., `https://ext-my.demisto.live/xsoar/instance/execute/teams`.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:63]:
```suggestion
For Cortex XSOAR version 6.x: `<CORTEX-XSOAR-URL>/instance/execute/<INTEGRATION-INSTANCE-NAME>`, e.g., `https://my.demisto.live/instance/execute/teams`.

For Cortex XSOAR version 8: `https://ext-<CORTEX-XSOAR-Domain>/xsoar/instance/execute/<INTEGRATION-INSTANCE-NAME>`, e.g., `https://ext-my.demisto.live/xsoar/instance/execute/teams`.
```

File: Packs/MicrosoftTeams/ReleaseNotes/1_4_43.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Microsoft Teams
+
+- Added support for *XSOAR 8*.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftTeams/ReleaseNotes/1_4_43.md:6]:
```suggestion
- Added support for Cortex XSOAR 8.
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md [hunk 3]
```diff
@@ -56,18 +56,21 @@ In order to verify that the messaging endpoint is open as expected, you can surf
...
 
 The integration instance name, `teams` in this example, needs to be configured in the [Configure Microsoft Teams on Cortex XSOAR](#configure-microsoft-teams-on-cortex-xsoar) step.
 
 The port to be configured in [Configure Microsoft Teams on Cortex XSOAR](#configure-microsoft-teams-on-cortex-xsoar) step should be any available port that is not used by another service.
 
-In addition, make sure ***Instance execute external*** is enabled.
+In addition, make sure ***Instance execute external*** is enabled, (for xsoar version less than 8.0.0).
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:69]:
```suggestion
In addition, make sure ***Instance execute external*** is enabled (for Cortex XSOAR 6.x).
```
What is ***Instance execute external**?  Is it a command?  a parameter?  an argument?  I could not find it in the yml file.

**Comment by ShirleyDenkberg** [comment]
@thefrieddan1 Doc review completed.

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 4]
```diff
@@ -292,7 +310,8 @@ def process_incident_create_message(demisto_user: dict, message: str, request_bo
         server_links: dict = demisto.demistoUrls()
         server_link: str = server_links.get('server', '')
         data = f"Successfully created incident {created_incident.get('name', '')}.\n" \
-               f"View it on: {server_link}/#/WarRoom/{created_incident.get('id', '')}"
+               f"View it on: {server_link + '/#' if not is_demisto_version_ge('8.0.0') else server_link}/" \
```
Comment (thefrieddan1) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:313]: I would do the server_link assignment in a line before 312
server_link = server_link + '/#' if not is_demisto.... else server_link.
and in line 313 leave it as it was

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 5]
```diff
@@ -2141,7 +2163,7 @@ def mirror_investigation():
         service_url: str = integration_context.get('service_url', '')
         server_links: dict = demisto.demistoUrls()
         server_link: str = server_links.get('server', '')
-        warroom_link = f'{server_link}/#/WarRoom/{investigation_id}'
+        warroom_link = f"{server_link + '/#' if not is_demisto_version_ge('8.0.0') else server_link}/WarRoom/{investigation_id}"
```
Comment (thefrieddan1) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:2166]: same here consider setting the server_link with /# in a line before 2166. 
This line as it is now is not very readable

**Review by thefrieddan1 — APPROVED** [review]
Do not merge until there is no approval in slack from @michal-dagan / @yucohen 
Please notice the comments about

**Pull request merged**